engine/dlib/src/test/test_buffer.cpp

```diff
@@ -287,6 +292,8 @@ TEST_F(BufferTest, ValidateBuffer)
     r = dmBuffer::ValidateBuffer(buffer);
     ASSERT_EQ(dmBuffer::RESULT_GUARD_INVALID, r);
 
+    memcpy(ptr, old_bytes, 4);
```

Comment:
Triggered by Asan's LeakDetection.

The buffer tests where testing memory corruption.
Now I restore the buffers afterwards, so that they can be successfully deleted.
These are old tests, so I guess newer clang caught this 🤔